core/src/style/polylineStyle.cpp

```diff
@@ -214,38 +214,37 @@ void PolylineStyle::buildMesh(const Line& _line, Parameters& _params, VboMesh& _
             vertices.push_back({position, order, texCoord, normal, { width, slope }, color});
         },
         [&](size_t sizeHint){ vertices.reserve(sizeHint); },
-        _params.fill.cap,
-        _params.fill.join,
+        _params.outline.cap,
+        _params.outline.join,
         _params.keepTileEdges
     };
 
-    Builders::buildPolyLine(_line, builder);
-
     if (_params.outlineOn && (_params.outline.width > 0.f || _params.outline.slope > 0.f)) {
+        Builders::buildPolyLine(_line, builder);
+    }
 
-        // Must update existing variables, they are captured (and used) by the builder
-        color = _params.outline.color;
-        width += _params.outline.width;
-        slope += _params.outline.slope;
-        order = std::min(_params.outline.order, _params.fill.order) - 0.5f;
-
-        if (_params.outline.cap != _params.fill.cap || _params.outline.join != _params.fill.join) {
-            // need to re-triangulate with different cap and/or join
-            builder.cap = _params.outline.cap;
-            builder.join = _params.outline.join;
-            Builders::buildPolyLine(_line, builder);
-        } else {
-            // re-use indices and positions from original line
-            size_t oldSize = builder.indices.size();
-            size_t offset = vertices.size();
-            builder.indices.reserve(2 * oldSize);
-
-            for(size_t i = 0; i < oldSize; i++) {
-                builder.indices.push_back(offset + builder.indices[i]);
-            }
-            for (size_t i = 0; i < offset; i++) {
-                vertices.push_back({ vertices[i], order, { width, slope }, color });
-            }
+    // Must update existing variables, they are captured (and used) by the builder
+    color = _params.fill.color;
+    width = _params.fill.width;
+    order = _params.fill.order;
+    slope = _params.fill.slope;
+
+    if (!_params.outlineOn || _params.outline.cap != _params.fill.cap || _params.outline.join != _params.fill.join) {
```

Comment:
A simpler change would be to keep the original order. Before processing the outline move out the builder indices to, say, fillIndices and vertices to fillVertices. At the end add the meshes in the intended order

```
mesh.addVertices(std::move(vertices), std::move(builder.indices));
mesh.addVertices(std::move(fillVertices), std::move(fillIndices));
```